### ISSUE TITLE
convert_read fail: TypeError: put() got an unexpected keyword argument 'expectedrows'

### ISSUE THREAD

**Issue by solegalli** [nilmtk/nilmtk#858]
Hi team, could I get help with the below error?

![Screen Shot 2020-02-27 at 16 58 01](https://user-images.githubusercontent.com/11805379/75461251-c7260680-5982-11ea-85e5-384d7a39ac1e.png)

**Comment by orestislampridis** [comment]
I was about to post the same error! I get exactly the same error. Can anyone help?

**Comment by solegalli** [comment]
Hey all, @orestislampridis, apparently, this issue occurs only with the package that we install from conda. Installing the pakage from the github repo resolves the issue.

If you created an environment as suggested in the nilmtk docs, then uninstall niltmk:
`conda uninstall nilmtk`

Next, I did the following:
1) fork the repo
2) clone fork into my local
3) cd into clone
4) `pip install -e .` (so i can modify)

This will still not run, need to add:
5) `pip install 'git+https://github.com/nilmtk/nilm_metadata.git#egg=nilm_metadata'`

Alternatively, see instructions here:
https://github.com/nilmtk/nilmtk/blob/cfafedca3420ea3e7fa2a4c78910f41264abc894/.travis.yml

**Comment by Oussama1920** [comment]
i have the same Issue !!! 
I Tried to reinstall from github repo but i'm in the same problem !

**Comment by simplyoptimisticonly** [comment]
@solegalli I was also getting the same error that you were getting i.e. expectedrow

But then I followed what you mentioned i.e. uninstall nilmtk in Anaconda followed by git clone etc. but I'm still getting the same error.

So can you pls share the redd.h5 if you are able to proceed further. I've tried on both Windows 10 and Ubuntu but the same error is coming.

**Comment by PMeira** [comment]
I uploaded a new conda package (version 0.4.1) with constrained dependencies. Try updating or reinstalling via conda first.

We don't recommend installing via pip in a conda setup as pip is not fully aware of conda packages and ends up corrupting everything.

Probably downgrading Pandas to 0.25.3 is enough though. Beware that using Pandas 1.0+ without fixing several other issues might end in unexpected/wrong results.

There might be some other package conflict besides Pandas, so feel free to open issues this kind of errors. I hope we upgrade everything by the end of month.

**Comment by simplyoptimisticonly** [comment]
@PMeira thx... I updated my Conda Environment as follows i.e. from the yml file of commit [1e2727e](https://github.com/nilmtk/nilmtk/commit/1e2727ece4e4d881e1aba8622a1bd069f4d5ba19)

conda env update --prefix C:\Users\Calvin\.conda\envs\nilmtk-env --file environment.yml  --prune

After that, when I tried the REDD and the issue disappeared... :)

Loading house 1... 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 
Loading house 2... 1 2 3 4 5 6 7 8 9 10 11 
Loading house 3... 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 
Loading house 4... 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 
Loading house 5... 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 
Loading house 6... 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 
Loaded metadata
Done converting YAML metadata to HDF5!
Done converting REDD to HDF5!
